src/main/java/com/amplitude/api/IdentifyInterceptor.java

```diff
@@ -206,4 +214,36 @@ private boolean isActionOnly(JSONObject event, String action) {
     private long saveIdentifyProperties(JSONObject event) {
         return dbHelper.addIdentifyInterceptor(event.toString());
     }
+
+    private boolean isIdentityUpdated(JSONObject event) {
+        try {
+            if (!identitySet.getAndSet(true)) {
+                userId = event.getString("user_id");
+                deviceId = event.getString("device_id");
+                return true;
+            }
+            boolean isUpdated = false;
+            if (isIdUpdated(userId, event.getString("user_id"))) {
+                userId = event.getString("user_id");
+                isUpdated = true;
+            }
+            if (isIdUpdated(deviceId, event.getString("device_id"))) {
+                deviceId = event.getString("device_id");
+                isUpdated = true;
+            }
+            return isUpdated;
+        } catch (JSONException e) {
+            return false;
```

Comment:
Default to true?

Reply:
yeah, I guess true is safer. Let me update it.